### DISCUSSION TITLE
Bookmark Create Dates not Synced Unfortunately

### DISCUSSION THREAD

**Discussion by tockschmock** [floccusaddon/floccus#1437] [Q&A]
I tried the extension with uploading a BM collection and downloading it into another account.
Unfortunately the Create Dates of the BMs did not sync.

The downloaded BMs received the date of the download instead of the original Create Date.

Is there a way to preserve the original creation date of each bookmark (like an export/import does) ?

**Comment by tockschmock** [comment]
![2023-12-02   19-10-36](https://github.com/floccusaddon/floccus/assets/68165681/d84f4bba-39d4-4900-994c-c497eb73d520)
A sequence:
- research A with BMs A1, A2, A3,   XXX  A4, A5
- research B with BMs B1, B2, XXX  B3, B4, B5
- search C with BMS C1, C2, XXX  C3, C4.C5

and XXX multiple floccus syncs

results in a synced BM order of

C3
C4
C5

B3
B4
B5
C1
C2

A4
A5
B1
B2

A1
A2
A3

with all blocks having the same time (of their respective sync actions).

I think it is key, to preserve the CreateDates of the bookmarks during the sync, to achieve a meaningful order on the sync target..

Otherwise a great extension ...

**Comment by tockschmock** [comment]
the problem is also mentioned here:
[https://github.com/floccusaddon/floccus/issues/1148#issue-1221924016](url)

_If I sort my bookmarks in Android it would be nice that, after syncing in Firefow desktop, the new sorting order appears._

**Reply by marcelklehr** [reply]
This is unrelated also. The sort order in Android is only a view, if you want to display bookmarks as you've sorted them in other browsers, you need to use the custom sort order option, but it's currently not possible to change the custom sort order in the Android app.

**Comment by tockschmock** [comment]
and here:
`https://github.com/floccusaddon/floccus/issues/1269#issuecomment-1279934284`

_It looks as if the order of the folders corresponds to the time they were created. 
The last folder created is displayed last in the directory tree._

**Reply by marcelklehr** [reply]
This is unrelated.

**Comment by marcelklehr** [comment]
> Is there a way to preserve the original creation date of each bookmark?

No, this is not possible.

> I think it is key, to preserve the CreateDates of the bookmarks during the sync, to achieve a meaningful order on the sync target

I do not follow you. CreateDate cannot be synced, but the bookmark order does get synced.

**Comment by tockschmock** [comment]
Is there a technical problem to sync the creation dates along the bookmarks name, URL and position?
I am not a programmer, but assume the creation date of a bookmark is just some more bytes.
There are Chrome extensions, exporting branches of the bookmark tree including their creation dates, so I assume, they are easily accesible.
If the data is there, it should not be lost in thy sync independend of its current value or usage potential in my opinion.

For me the creation date is very important for most of my bookmarks except the few regions, I prefer having custom ordered  (the bookmarks bar / the subfolders in the bookmarks bar).
All Other bookmarks and its subfolders are sorted from newest to oldest with the help of an extension (Sprucemarks). Later I split them into months folders and subject folders for a little organization.
If bookmarks are imported from another account/machine, they should fit into this chronological order.
The chronological sorting extension (Sprucemarks) also heals Chromes and most browsers ridiculus default, to sort new bookmarks to the END of the bookmarks folder lists, which makes the bookmarks unusable on smaller mobile devices  Larger bookmark folders require endless scrolling down to see a most recent bookmark here ... 
I believe 98% of users would prefer a chronological order of their bookmarks "Newest to Oldest" as the default and proposed this to chromium.org here:
[https://bugs.chromium.org/p/chromium/issues/detail?id=1443217#c_ts1683529906](url)

Unfortunately Chrome does not support using the creation dates and I have to resort to extensions that recognize, filter and sort creation dates:
[https://chromewebstore.google.com/detail/bookmark-manager-plus/pfbeenngglcojppheoegjjjomfkejibg?pli=1](url)
for my organization.

Edge also recently introdced a sidebar that allows sorting by date, so there is small progress.
floccus imported bookmarks would appear in completely inconsistent date regions (the sync dates) here ...

Bottom line:
Although probably only a few experts currently use and/or take care of the creation date of their bookmarks, it should be preserved in the sync process.
For me personally, a sync loosing the creation dates is useless.

I am happy to support with more information and tests for this otherwies great extension.

**Reply by marcelklehr** [reply]
> Is there a technical problem to sync the creation dates along the bookmarks name, URL and position?

It is possible to transmit the creation date of course, but there is no API to set the creation date on bookmarks in the browser.

**Comment by tockschmock** [comment]
transmitting the date is all I need and recommend

please explain the second part of your comment (_no API to set the creation date on bookmarks in the browser_)
why do you miss an API?

the date is created the moment, I save a bookmark
if it "survives" the sync, experts working with it and the new Edge sidebar using it will be happy 

the extension cited above displays the creation dates and allows sorting, grouping  and filterimg with dates

**Reply by marcelklehr** [reply]
> transmitting the date is all I need and recommend

transmitting will not get you anywhere. That's like saying you can found a company just by using the postal service. Sure, the postal service is necessary, but you need someone on the other end to receive your letters, too.

> why do you miss an API?

Because the API is not there. Please simply believe me. I am not being stubborn, I am just genuinely answering your question. It is not possible. It may be possible for extensions to display the creation date, but to my knowledge it is not possible to override the creation date when creating bookmarks. You can bargain and plead with me, but I have no power over this.
